### PR TITLE
Add docstring to add_recording

### PR DESCRIPTION
**Pull request by h-mayorquin**
`add_recording` is not showing up in the API somehow. I am wondering if it is the lack of docstring...

anyway, a docstring is good to add anyway so adding it here.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
This does add `add_recording` to the API. So it seems that the auto-docs only works when you have a docstring.

**Comment by CodyCBakerPhD** [comment]
> This does add add_recording to the API. So it seems that the auto-docs only works when you have a docstring.

Yeah, we've known that for a while

You should have a chat with @pauladkisson sometime; we solved this problem on ROIExtractors by creating a custom pre-commit that ensures all publically exposed functions and classes have docstrings (which include from inheritance)

**Comment by h-mayorquin** [comment]
I don't really like that solution : P

I remember finding the pre-commit annoying in roi-extractors. Makes me less likely to want to contribute. It was something about enforcing spaces that did not make a lot of sense to me.

I will have to thing again.

**Comment by CodyCBakerPhD** [comment]
I think you're talking about a separte pre-commit for pydocstyle which ensures strict numpy style compliance

The check for ensuring public functions have a non-empty docstring is its own thing (so that `help(< some method >)` or `< some method >?` doesn't return `< no docstring >`) and even if annoying is valuable for both the API docs (which chatbots can then scrub) and general usage

Have you tried writing docstring using GitHub copilot? For me it formats everything perfectly, takes less than 15 seconds per method, and gets it almost perfect in the descriptions. So nowadays I don't think the cost is high at all to enforce

**Comment by pauladkisson** [comment]
> we solved this problem on ROIExtractors by creating a custom pre-commit that ensures all publically exposed functions and classes have docstrings (which include from inheritance)

Actually, on roiextractors it's a custom github workflow (not pre-commit) that ensures all publicly exposed functions and classes have docstrings.  So, it just has to be satisfied before merging, which is really not much of a burden at all.

**Comment by h-mayorquin** [comment]
I stand corrected, thanks for explaining it to me guys. I will check it out.

**Comment by bendichter** [comment]
I think it makes sense to enforce that all functions and methods must either be marked as private with a `_` or have a docstring, especially since it appears this is critical for autodoc to work. 

For modules, I have found it annoying and often repetitive to need to create docstrings for basic `__init__.py` files. Would it be possible to omit these?

(duplicated here: https://github.com/catalystneuro/.github/issues/17)

File: src/neuroconv/tools/spikeinterface/spikeinterface.py [hunk 1]
```diff
@@ -924,6 +921,60 @@ def add_recording(
...
+        Key in the metadata dictionary containing metadata information for the specific ElectricalSeries.
+    write_electrical_series : bool, default=True
+        If True, writes the ElectricalSeries to the NWBFile. If False, no ElectricalSeries is written.
+    write_scaled : bool, default=False
+        If True, writes the traces in microvolts (uV) with the appropriate conversion.
+        If False, the data is stored as-is, and the correct conversion factors are added to the NWBFile.
+    iterator_type : {'v2', None}, default='v2'
+        The type of DataChunkIterator to use when writing data in chunks. Options are:
+
+        - 'v1': The original DataChunkIterator from hdmf.data_utils.
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:963]: I think we deprecated iterator type completely, didn't we? @h-mayorquin 

Or am I just thinking about the ophys submodule

If not, given how stable the generic iterator has proven over the last 2 years I think we can go all in on that now (it plays a lot better with the backend configuration tooling too)

**Comment by CodyCBakerPhD** [comment]
> For modules, I have found it annoying and often repetitive to need to create docstrings for basic __init__.py files. Would it be possible to omit these?

See https://github.com/catalystneuro/.github/issues/17#issuecomment-2291605756 part 2 for full details

But in summary, if we do a better job of controlling what we publically expose as submodules then we can indeed ignore the majority of `__init__.py` files once their parent submodules are marked as private

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD @bendichter 
> Yeah, we've known that for a while

> I think it makes sense to enforce that all functions and methods must either be marked as private with a _ or have a docstring, especially since it appears this is critical for autodoc to work.

Not that it has any bearing on what is desirable but note that with the changes in #1011 functions no longer require a docstring for autodoc to display then. That was a configuration option not a requirement.

https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-option-automodule-undoc-members

**Comment by bendichter** [comment]
@h-mayorquin thanks that's helpful to know. I still think we should require docstrings for non-private methods.

**Comment by h-mayorquin** [comment]
> @h-mayorquin thanks that's helpful to know. I still think we should require docstrings for non-private methods.

Agree, will check out the solution of roiextractors mentioned above.

**Comment by h-mayorquin** [comment]
So, is the docstring OK, can we merge this?

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**